File: src/cqrs/mr/commandhandlers/CreateInventoryItemHandler.java
package cqrs.mr.commandhandlers;

import cqrs.core.Handler;
import cqrs.core.Repository;
import cqrs.mr.commands.CreateInventoryItem;
import cqrs.mr.domain.InventoryItem;

public class CreateInventoryItemHandler implements Handler<CreateInventoryItem>{
	private final Repository<InventoryItem> repository;
	
	public CreateInventoryItemHandler(Repository<InventoryItem> repository) {
		this.repository = repository;
	}

	@Override
	public void handle(CreateInventoryItem command) {
		InventoryItem item = new InventoryItem(command.id, command.name);
		repository.save(item, -1);
	}
}

File: app/cqrs/mr/commandhandlers/CreateInventoryItemHandler.scala
package cqrs.mr.commandhandlers

import cqrs.core.Handler
import cqrs.core.Repository
import cqrs.mr.commands.CreateInventoryItem
import cqrs.mr.domain.InventoryItem

 class CreateInventoryItemHandler(repository: Repository[InventoryItem]) extends Handler[CreateInventoryItem] {
	def handle(command: CreateInventoryItem): Unit = {
		val item = new InventoryItem(command.id, command.name)
		repository.save(item, -1)
	}
}
